apps/files/src/store/renaming.ts

```diff
@@ -8,11 +8,77 @@ import type { RenamingStore } from '../types'
 import axios, { isAxiosError } from '@nextcloud/axios'
 import { emit, subscribe } from '@nextcloud/event-bus'
 import { NodeStatus } from '@nextcloud/files'
+import { DialogBuilder } from '@nextcloud/dialogs'
 import { t } from '@nextcloud/l10n'
-import { basename, dirname } from 'path'
+import { basename, dirname, extname } from 'path'
 import { defineStore } from 'pinia'
 import logger from '../logger'
 import Vue from 'vue'
+import IconCancel from '@mdi/svg/svg/cancel.svg?raw'
+import IconCheck from '@mdi/svg/svg/check.svg?raw'
+
+let isDialogVisible = false
+
+const showWarningDialog = (oldExtension: string, newExtension: string): Promise<boolean> => {
+	if (isDialogVisible) {
+		return Promise.resolve(false)
+	}
+
+	isDialogVisible = true
+
+	let message
+
+	if (!oldExtension && newExtension) {
+		message = t(
+			'files',
+			'Adding the file extension "{new}" may change how the file is handled.',
+			{ new: newExtension },
+		)
+	} else if (!newExtension) {
+		message = t(
+			'files',
+			'Removing the file extension "{old}" may result in unexpected behavior.',
+			{ old: oldExtension },
+		)
+	} else {
+		message = t(
+			'files',
+			'Changing the file extension from "{old}" to "{new}" may affect how the file is handled.',
+			{ old: oldExtension, new: newExtension },
+		)
+	}
```

Comment:
```suggestion
	if (!oldExtension && newExtension) {
		message = t(
			'files',
			'Adding the file extension "{new}" may change how the file is handled.',
			{ new: newExtension },
		)
	} else if (!newExtension) {
		message = t(
			'files',
			'Removing the file extension "{old}" may result in unexpected behavior.',
			{ old: oldExtension },
		)
	} else {
		message = t(
			'files',
			'Changing the file extension from "{old}" to "{new}" may affect how the file is handled.',
			{ old: oldExtension, new: newExtension },
		)
	}
```

These are more generic, but I leave it to you @marcoambrosini